src/components/WalletModal/index.tsx

```diff
@@ -135,46 +111,13 @@ export default function WalletModal({ openSettings }: { openSettings: () => void
     getConnection,
   ])
 
-  // Used to track the state of the drawer in async function
+  // Used to provide proper state of drawer to callbacks triggered asynchronously
   const drawerOpenRef = useRef(drawerOpen)
   drawerOpenRef.current = drawerOpen
 
-  const tryActivation = useCallback(
-    async (connection: Connection) => {
-      // Skips wallet connection if the connection should override the default behavior, i.e. install metamask or launch coinbase app
-      if (connection.overrideActivate?.()) return
-
-      // log selected wallet
-      sendEvent({
-        category: 'Wallet',
-        action: 'Change Wallet',
-        label: connection.type,
-      })
-
-      try {
-        setPendingConnection(connection)
-        setPendingError(undefined)
-
-        await connection.connector.activate()
-        console.debug(`connection activated: ${connection.getName()}`)
-        dispatch(updateSelectedWallet({ wallet: connection.type }))
-        if (drawerOpenRef.current) toggleWalletDrawer()
-      } catch (error) {
-        console.debug(`web3-react connection error: ${JSON.stringify(error)}`)
-        // TODO(WEB-3162): re-add special treatment for already-pending injected errors
-        if (didUserReject(connection, error)) {
-          setPendingConnection(undefined)
-        } else {
-          setPendingError(error)
-
-          sendAnalyticsEvent(InterfaceEventName.WALLET_CONNECT_TXN_COMPLETED, {
-            result: WalletConnectionResult.FAILED,
-            wallet_type: connection.getName(),
-          })
-        }
-      }
-    },
-    [dispatch, setPendingError, toggleWalletDrawer]
+  const activate = useCallback(
```

Comment:
Do we need `useCallback` here? Can't we just pass it as an arrow function? Looks a bit unnecessary, given that we still pass an arrow function in line `133`. If we removed this, we would get rid of two arrow functions and just use one (pass everything inline on line `133`)

Reply:
Great point